python-frontend/src/main/java/org/sonar/python/lexer/IPynbCellDelimiterChannel.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.lexer;
+
+import com.sonar.sslr.api.Token;
+import com.sonar.sslr.impl.Lexer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.sonar.python.api.PythonTokenType;
+import org.sonar.sslr.channel.Channel;
+import org.sonar.sslr.channel.CodeReader;
+
+public class IPynbCellDelimiterChannel extends Channel<Lexer> {
+
+  private final StringBuilder tmpBuilder = new StringBuilder();
```

Comment:
it is a personal preference but in my opinion, if we have a constructor, it is better to initialize all the fields in it, to have a common place for initialization.

Reply:
That's a fair point, and I must admit I don't have a strong opinion on it. Though in this case to be honest I just went with consistency with other lexer channels (like `IndentationChannel`, `StringLiteralsChannel`, etc...)